theta_dup/include/theta_sketch_dup.h

```diff
@@ -0,0 +1,2209 @@
+   * sketch
+   * @return an instance of the sketch
+   */
+  static compact_theta_sketch_dup_alloc<A> deserialize(
+      const void* bytes, size_t size, uint64_t seed = DEFAULT_SEED);
+
+  /**
+   * @return true if *this equals r
+   */
+  virtual bool isEqual(const compact_theta_sketch_dup_alloc& r) const;
+
+ private:
+  typedef typename std::allocator_traits<A>::template rebind_alloc<uint64_t>
+      AllocU64;
+
+  vector_u64<A> keys_;
+  uint16_t seed_hash_;
+  bool is_ordered_;
+
+  friend theta_sketch_dup_alloc<A>;
+  friend update_theta_sketch_dup_alloc<A>;
+
+  /* TODO: support set operations
+   * friend theta_union_alloc<A>;
+   * friend theta_intersection_alloc<A>;
+   * friend theta_a_not_b_alloc<A>;
+   */
+
+  compact_theta_sketch_dup_alloc(bool is_empty, uint64_t theta,
+                                 vector_u64<A>&& keys, uint16_t seed_hash,
+                                 bool is_ordered);
+  static compact_theta_sketch_dup_alloc<A> internal_deserialize(
+      std::istream& is, uint8_t preamble_longs, uint8_t flags_byte,
+      uint16_t seed_hash);
+  static compact_theta_sketch_dup_alloc<A> internal_deserialize(
+      datasketches_pb::ThetaSketchDup& pb, uint8_t preamble_longs, uint8_t flags_byte,
+      uint16_t seed_hash);
+  static compact_theta_sketch_dup_alloc<A> internal_deserialize(
+      const void* bytes, size_t size, uint8_t preamble_longs,
+      uint8_t flags_byte, uint16_t seed_hash);
+};
+
+// builder
+
+template <typename A>
+class update_theta_sketch_dup_alloc<A>::builder {
+ public:
+  static const uint8_t MIN_LG_K = 5;
+  static const uint8_t DEFAULT_LG_K = 12;
+  static const resize_factor DEFAULT_RESIZE_FACTOR = X8;
+
+  /**
+   * Creates and instance of the builder with default parameters.
+   */
+  builder();
+
+  /**
+   * Set log2(k), where k is a nominal number of entries in the sketch
+   * @param lg_k base 2 logarithm of nominal number of entries
+   * @return this builder
+   */
+  builder& set_lg_k(uint8_t lg_k);
+
+  /**
+   * Set resize factor for the internal hash table (defaults to 8)
+   * @param rf resize factor
+   * @return this builder
+   */
+  builder& set_resize_factor(resize_factor rf);
+
+  /**
+   * Set sampling probability (initial theta). The default is 1, so the sketch
+   * retains all entries until it reaches the limit, at which point it goes into
+   * the estimation mode and reduces the effective sampling probability (theta)
+   * as necessary.
+   * @param p sampling probability
+   * @return this builder
+   */
+  builder& set_p(float p);
+
+  /**
+   * Set the seed for the hash function. Should be used carefully if needed.
+   * Sketches produced with different seed are not compatible
+   * and cannot be mixed in set operations.
+   * @param seed hash seed
+   * @return this builder
+   */
+  builder& set_seed(uint64_t seed);
+
+  /**
+   * This is to create an instance of the sketch with predefined parameters.
```

Comment:
means "default parameter"? if so, briefly list them here?

Reply:
Updated. See [this commit](https://github.com/googleinterns/cardinality-estimation-with-data-deletions/pull/2/commits/169209954c971e564bb735f71fcbcf3e1420a616).